### ISSUE TITLE
Button retains mdc-ripple-upgraded--background-focused style even though it's not focused

### ISSUE THREAD

**Issue by ndtreviv** [material-components/material-components-web-react#916]
I have three buttons that I'm using as a search results view type button bank (grid view, list view, map view).

When a particular view type is active, the corresponding `Button` is marked as `disabled` so the user can't reselect it (it's already been applied).

## Expected behaviour
When a user clicks an option, the button relating to that option should be highlighted, other buttons should lose their highlight.

## Actual behaviour
Initial load with list view active:
<img width="464" alt="Screenshot 2019-06-07 at 15 51 40" src="https://user-images.githubusercontent.com/1530653/59113087-4d557500-893c-11e9-97fc-546b10691f25.png">

I think that disabled is the wrong thing. What I really want is a a button group where the buttons work as a radio buttons (eg: https://getbootstrap.com/docs/4.1/components/buttons/#checkbox-and-radio-buttons)
So my first problem with this is that the colours are wrong. The selected one should be highlighted, and the others should be secondary in this case.

It doesn't feel like there's a paradigm for that at the moment...?

After first click:
<img width="458" alt="Screenshot 2019-06-07 at 15 51 50" src="https://user-images.githubusercontent.com/1530653/59113333-bc32ce00-893c-11e9-94f3-3f9a1708e91f.png">

The selected button retains its focus style, even if I attribute focus to a different element on the screen.

After second click:
<img width="465" alt="Screenshot 2019-06-07 at 15 51 57" src="https://user-images.githubusercontent.com/1530653/59113410-d53b7f00-893c-11e9-9b75-8ce9ceeac737.png">

The first click-button _still_ retains its focused style.

## Code
```
        <div className="search-tools__format">
          <Button icon={<MaterialIcon icon="map" />} disabled={format === "map"} onClick={e => {changeViewFormat("map");}} />
          <Button icon={<MaterialIcon icon="view_list" />} disabled={format === "list"} onClick={e => {changeViewFormat("list");}} />
          <Button icon={<MaterialIcon icon="view_module" />} disabled={format === "grid"} onClick={e => {changeViewFormat("grid");}} />
        </div>
```

**Comment by ndtreviv** [comment]
Could the state update onClick be interrupting the ripple somehow?

**Comment by TroyTae** [comment]
Can you check element's focus?
As you know, element's focus is removed when click other area.

Or if you share stackblitz or codepen, that is very helpful :)

**Comment by ndtreviv** [comment]
Workaround is to override the style and set the --background-focused version to have no background colour. The ripple still seems to work ok.

**Comment by ndtreviv** [comment]
> Can you check element's focus?
> As you know, element's focus is removed when click other area.
> 
> Or if you share stackblitz or codepen, that is very helpful :)

I'm literally clicking on other elements on the screen, highlighting text, causing state changes that means a neighbouring button get re-rendered, but it still retains "focus".

I've never used stackblitz before. I'll try and get a reproduceable setup going.

**Comment by ndtreviv** [comment]
I managed to get this far: https://stackblitz.com/edit/react-u9rd1h
But I can't work out how to import the @material scss into the styles.scss without it generating errors. :( Do you know how to do this?

**Comment by poying** [comment]
Same issue here.
![out](https://user-images.githubusercontent.com/1190072/59196387-1643c580-8bc1-11e9-985d-a858429493ed.gif)

**Comment by TroyTae** [comment]
@ndtreviv @poying 
Thank you both :)
Now I'm debugging this.

**Comment by poying** [comment]
@TroyTae [This change](https://github.com/poying/material-components-web-react/commit/7785a8e077093bf09b2a54c30ac40208052e6f7d) can solve the issue, but I am not sure whether it will cause other issues or not. I hope this is helpful for debugging.

**Comment by TroyTae** [comment]
@poying 
Thank you for showing your code :)

**Comment by TroyTae** [comment]
*TBD*
This issue is fired when element's disabled is being true after click.
After click event, element has `mdc-ripple-upgraded--background-focused` class.
But blur event isn't fired at disabled element, so this focus class is keep alive.
There is need to think about how to solve this problem.

**Comment by moog16** [comment]
@ndtreviv it looks like your [stackblitz](https://github.com/material-components/material-components-web-react/issues/916#issuecomment-500340609) is working as expected. How are you reproducing the original issue?

**Comment by ndtreviv** [comment]
@moog16 I can't get the stackblitz to import the styles properly. It just errors whenever I import the @material based scss files.

You can see them commented out in the style.scss file - if you can help me figure out how to get that bit working on the stackblitz then I'll get you a better reproduction of the problem.

It looks like @TroyTae has worked out what the problem could be:

> **TBD**
> This issue is fired when element's disabled is being true after click.
> After click, element has `mdc-ripple-upgraded--background-focused` class.
> But blur event isn't fired at disabled element, so this focus class is keep alive.
> There is need to think about how to solve this problem.

It sounds plausible. When a user clicks a button it becomes disabled on re-render so that they can't re-click the same button.

**Comment by TroyTae** [comment]
I made a PR to reproduce this issue.
That PR don't contain solving code!
I'm still think how to solve it...

**Comment by moog16** [comment]
@ndtreviv it looks like `@import '~@material/react-button/index.scss';` works but material-icon's sass is not. I can confirm that the sass file does [exist](https://unpkg.com/@material/react-material-icon@0.13.0/index.scss).

**Comment by moog16** [comment]
@TroyTae I'll look at the PR

**Comment by moog16** [comment]
@TroyTae thanks for creating that PR...I tried reproing it in MDC Web to be sure, and its definitely just a React issue.

**Comment by moog16** [comment]
fixed by #943

**Comment by adamsoffer** [comment]
@moog16 This seems to unfortunately still be an issue -> 

https://codesandbox.io/s/material-react-ripple-bug-muly9
